src/query/procedure/module.cpp

```diff
@@ -634,11 +656,9 @@ void RegisterMgDeleteModuleFile(ModuleRegistry *module_registry, utils::RWLock *
       return;
     }
 
-    const auto &module_directories = module_registry->GetModulesDirectory();
-    if (std::all_of(module_directories.begin(), module_directories.end(),
-                    [&](const auto &base_directory) { return !IsSubPath(base_directory, path); })) {
-      static_cast<void>(
-          mgp_result_set_error_msg(result, "The specified file isn't contained in one of the module directories."));
+    const auto parent_module_directory = ParentModuleDirectory(*module_registry, path);
```

Comment:
I mentioned the longest common prefix, because if somebody uses nested query module directories (both `a` and `a/b` is specified) then we might end up deleting an empty query module directory. Although I am not sure if that is a problem or not.